File: Plugins/Android/BlahSavesHelper.java
package com.blah.saves;

import android.app.Activity;
import android.content.Context;
import android.os.StatFs;

import java.io.File;
import java.io.IOException;

public class BlahSavesHelper
{
    public static String GetPath(Activity activity) throws IOException {
        Context context = activity.getApplicationContext();
        File dir = context.getFilesDir();
        return dir.getCanonicalPath();
    }

    public static float GetAvailableSpaceMB(Activity activity) {
        Context context = activity.getApplicationContext();
        File dir = context.getFilesDir();
        
        StatFs stat = new StatFs(dir.getPath());
        long bytesAvailable = 0;
        bytesAvailable = stat.getBlockSizeLong() * stat.getAvailableBlocksLong();
        return bytesAvailable / (1024.f * 1024.f);
    }
}

File: BlahSavesHelper.cs
using UnityEngine;

namespace BlahSaves
{
internal static class BlahSavesHelper
{
	public static string GetPath()
	{
#if UNITY_EDITOR
		return Application.persistentDataPath;
#elif UNITY_ANDROID
		return GetAndroidPath();
#else
		return Application.persistentDataPath;
#endif
	}

	public static float GetSpaceMb()
	{
#if UNITY_EDITOR
		return float.MaxValue;
#elif UNITY_ANDROID
		return GetAndroidSpaceMb();
#else
		return float.MaxValue;
#endif
	}


#if UNITY_ANDROID
	private static string GetAndroidPath()
	{
		using var player    = new AndroidJavaClass("com.unity3d.player.UnityPlayer");
		using var activity  = player.GetStatic<AndroidJavaObject>("currentActivity");
		using var context   = activity.Call<AndroidJavaObject>("getApplicationContext");
		using var filesDir  = context.Call<AndroidJavaObject>("getFilesDir");
		var       filesPath = filesDir.Call<string>("getCanonicalPath");
		return filesPath;
	}

	private static float GetAndroidSpaceMb()
	{
		using var player   = new AndroidJavaClass("com.unity3d.player.UnityPlayer");
		using var activity = player.GetStatic<AndroidJavaObject>("currentActivity");
		using var context  = activity.Call<AndroidJavaObject>("getApplicationContext");
		using var filesDir = context.Call<AndroidJavaObject>("getFilesDir");
		var    path     = filesDir.Call<string>("getPath");

		using var statsFs        = new AndroidJavaObject("android.os.StatFs", path);
		var       blockSize      = statsFs.Call<long>("getBlockSizeLong");
		var       blocksCount    = statsFs.Call<long>("getAvailableBlocksLong");
		long      bytesAvailable = blockSize * blocksCount;
		return bytesAvailable / (1024.0f * 1024.0f);
	}
#endif
}
}